ledger/streaming/api/src/test_utils/fetcher.rs

```diff
@@ -32,20 +32,15 @@ impl MockFetcher {
 }
 
 impl BlockFetcher for MockFetcher {
-    type Single = impl Future<Output = Result<BlockStreamComponents>>;
-    type Multiple = impl Stream<Item = Result<BlockStreamComponents>>;
+    type Single<'s> = impl Future<Output = Result<BlockStreamComponents>> + 's;
+    type Multiple<'s> = impl Stream<Item = Result<BlockStreamComponents>> + 's;
 
-    fn fetch_single(&self, index: BlockIndex) -> Result<Self::Single> {
+    fn fetch_single(&self, index: BlockIndex) -> Self::Single<'_> {
         let result = self.results[index as usize].clone();
-        Ok(async { result })
+        async { result }
```

Comment:
Returning async block futures means that we'll have to pin up and down the line of streams given async blocks are fundamentally of type Unpin.

The design decision we face there is 
a) Does pinning via something like Box::Pin make things any slower or is it fine to do so?
b) If it is unacceptably slower or we could get a reasonable speedup by not requiring box::pin, we need a construction in the chain of streams to do so. That could be either having UnPin trait on our streams OR find a way to ensure our return types are producing explicit types that implement unpin.